### DISCUSSION TITLE
Is scaling to zero while there is ongoing traffic on kafka topic the expected behavior?

### DISCUSSION THREAD

**Discussion by gunniwho** [kedacore/keda#2594] [Q&A / Need Help] [answered]
I have a scaled object with a kafka trigger. The service is consuming a topic that has intermittent but steadily ongoing traffic on it, maybe a few messages per minute. It takes a very short time to process each message so every time keda polls, there is no lag on the topic. So every time, 5 minutes (cooldown period) from the time the service is scaled up to 1, it is scaled back down to 0 again. This doesn't seem intuitive to me. Shouldn't the trigger check if the offsets it is receiving when polling are the same as they were the last time or something like that? 
Thanks!

**Comment by zroubalik** [comment, marked as answer]
KEDA currently doesn't keep any state (this might be changed in the future https://github.com/kedacore/keda/issues/2282).

You can try to twen [the scaling behavior](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#configurable-scaling-behavior) or set min replicase to `1`.
